### PR TITLE
Visualizer: Add semantics editor

### PR DESCRIPTION
**Pull request by lilyx**
Adding semantics editor to the visualizer. Hope this one makes review easier :)

### CODE REVIEW

**Comment by pdubroy** [comment]
Thanks Meixian, that's perfect! I think Alex is going to take a look at this with you today, and I'll follow up tomorrow with my comments.

File: visualizer/parseTree.js [hunk 1]
```diff
@@ -156,6 +159,31 @@
     }
   }
 
+  // Hides or shows the semantic editor of `el`, which is a div.pexpr.
```
Comment (alexwarth) [inline visualizer/parseTree.js]: s/semantic/semantics

File: visualizer/parseTree.js [hunk 2]
```diff
@@ -156,6 +159,31 @@
...
+  function toggleSemanticEditor(el) {
```
Comment (alexwarth) [inline visualizer/parseTree.js]: Please add a comment that explains what this does, in a way that someone using it would understand.

**Comment by pdubroy** [comment]
I think this is out of date, right?

**Pull request closed without merging**